### ISSUE TITLE
[Inductor] Register pattern match with dropout in torchao as a customized pass

### ISSUE THREAD

**Issue by Valentine233** [pytorch/pytorch#143363]
### Description

In https://github.com/pytorch/ao/pull/1372, we try to add a pattern match pass for int8 SDPA in torchao. For the pass in pytorch, the order is `joint_custom_pre -> remove_noop_ops -> constant_folding -> pattern_matcher -> fallback_random -> joint_custom_post`. As we know, the clone() from dropout would be removed by `remove_noop_ops`. If we register the pattern as a `joint_custom_pre_pass`, there is still the clone() when `joint_custom_pre`, causing the int8 SDPA pattern match failure. If we register the pattern as a `joint_custom_post_pass`, it would firstly hit `_partial_softmax_pattern` which changes the decomposed ops in softmax, also causing the match failure. As a result, the best position to put the int8 SDPA pattern match is the same as `pattern_matcher` in pytorch, but we haven't got such a choice for customized pass. Is it possible to move the position of `joint_custom_pre` or create a new customized pass?

### Versions

pytorch: main branch.
torchao: main branch.

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @aakhundov @eellison @drisspg @Chillee @jerryzh168 @leslie-fang-intel

**Comment by jansel** [comment]
I'd be fine moving `joint_custom_pre` after``remove_noop_ops`/`constant_folding`.
